### PR TITLE
feat(linux): Dockerfile for linux builder

### PR DESCRIPTION
**Pull request by srl295**
- add Dockerfile in linux/Dockerfile, referencing linux/debian/control
- add docs in docs/build/linux-ubuntu.md

Fixes: #7102

@keymanapp-test-bot skip

### CODE REVIEW

**Comment by srl295** [comment]
I used the `keymanapp/` org found at https://hub.docker.com/u/keymanapp but there's no automation to push this build there.

I use this image a lot. It ought to be able to build all of keyman, But I've only tested it with core.

File: docs/build/linux-ubuntu.md [hunk 1]
```diff
@@ -64,3 +64,23 @@ All dependencies are already installed if you followed the instructions under [P
...
+docker build . -t keymanapp/keyman-linux-builder:latest
+```
+
+Once the image is built, it may be used to build parts of Keyman.
+
+```shell
+cd ../core
+# keep linux build artifacts separate
+mkdir -p build/linux
+docker run -it --rm -v $(pwd)/..:/home/build -v $(pwd)/build/linux:/home/build/core/build keyman-linux-builder:latest bash -c 'cd core; bash build.sh -d'
```
Comment (ermshiperete) [inline docs/build/linux-ubuntu.md:85]: Do we need both `-v`s? I'd think the latter is included in the former.

**Review by ermshiperete — APPROVED** [review]
:+1:

File: docs/build/linux-ubuntu.md (hunk 1 above)
Comment (srl295) [reply docs/build/linux-ubuntu.md:85]: no, by default it builds into `build/arch` which would stomp on your local build.   This creates a `build/linux` which is its own `build` subdirectory.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 16.0.53-alpha](https://keyman.com/downloads/releases/alpha/16.0.53)

**Comment by mcdurdin** [comment]
Just thinking about this -- I think the Dockerfile should be in /resources/devbox/linux, as ideally we will have multiple Docker images for different platform builds -- one for each platform (although Windows will probably be a non-starter for a while due to Delphi). And this is about build environment rather than something specific to Keyman.

Currently discussing whether or not we can do a Docker image for macOS builds as well. We may use this for future macOS build agents.

**Comment by srl295** [comment]
@mcdurdin OK, want a new PR to move it there?

**Comment by mcdurdin** [comment]
> @mcdurdin OK, want a new PR to move it there?

Oh yes please :grin:

**Comment by srl295** [comment]
> > @mcdurdin OK, want a new PR to move it there?
> 
> Oh yes please 😁

A wrinke here is that the Dockerfile depends on `linux/debian/control` - it actually is using the debian control file to pull in the deps.  Could have a script in `devbox` that copies over `linux/debian/control` into a gitignored location?

**Comment by mcdurdin** [comment]
It's not possible to `ADD ../../../linux/debian/control /tmp/control`?

**Comment by srl295** [comment]
> It's not possible to `ADD ../../../linux/debian/control /tmp/control`?

No, it's not possible to ADD outside of the docker context.  You could start from up three levels, but then it would send the entre repo as context.

**Comment by srl295** [comment]
conversely, could a script in devbox do `docker build ../../../linux` ?

**Comment by mcdurdin** [comment]
Yeah I guess so? Then we'll eventually end up with a Dockerfile in each of the primary top-level platform folders -- android/Dockerfile, developer/Dockerfile, ios/Dockerfile, linux/Dockerfile, mac/Dockerfile, web/Dockerfile, windows/Dockerfile?

Alternatively, just three: linux/Dockerfile, windows/Dockerfile, mac/Dockerfile, which contain all the toolchains needed for building any product that can be built on that platform. (Gosh what a wonderful world it would be if macOS would (legally) run in a VM on non-mac hardware...)

Windows is complicated because Delphi. macOS will be a massive image because XCode. So not sure this is yet ideal but step-by-step we improve our build environments I hope.

**Pull request merged**